### PR TITLE
Ontario rewrite

### PR DESCRIPTION
**Pull request by parkern342**

### CODE REVIEW

File: backend/ontario/ontario/__init__.py [hunk 1]
```diff
@@ -41,41 +46,59 @@ def __init__(self, context: ImageContext):
...
-        node.set_property("operation", "gegl:load")
-        node.set_property("path", path)
-        self.__parent.add_child(node)
+        # get file extention
+        split_ext: tuple = os.path.splitext(path)
+        file_ext: str = split_ext[-1]
+        file_ext = file_ext.lower()
+
+        # add node as a child to the image context
+        if file_ext == '.png':
```
Comment (notgull) [inline backend/ontario/ontario/__init__.py:59]: Ontario-Web currently stores images as webp, is there a way to support webp here or should I move to one of these formats?

File: backend/ontario/ontario/__init__.py [hunk 2]
```diff
@@ -99,29 +122,21 @@ def composite(self, other: "ImageBuilder") -> "ImageBuilder":
...
-        node = Gegl.Node()
-        node.set_property("operation", "gegl:over")
-        self.__parent.add_child(node)
-
-        # Connect the last node to the save node.
-        self.__nodes[-1].connect_to("output", node, "aux")
-        other.__nodes[-1].connect_to("output", node, "input")
-
-        self.__nodes.append(node)
+        # TODO: implement gegl:layer operation
```
Comment (notgull) [inline backend/ontario/ontario/__init__.py:125]: Could this be an exception? Composite is currently used in the Ontario-Web tests and I'd like this tests to error is this is unimplemented.

File: gimp/pictonode-gimp-plugin/GIMP 2.99/pictonode/custom_nodes.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+# implement gegl node operations using ontario backend
+
+class ImgSrcNode(GtkNodes.Node):
+    __gtype_name__ = 'SrcNode'
+
+    def __init__(self, *args, **kwds) -> None:
+        super().__init__(*args, **kwds)
+
+        # build an image contexts 
+        # add nodes to it
```
Comment (notgull) [inline gimp/pictonode-gimp-plugin/GIMP 2.99/pictonode/custom_nodes.py:74]: Did the file get cut off here?

File: backend/ontario/ontario/__init__.py (hunk 1 above)
Comment (parkern342) [reply backend/ontario/ontario/__init__.py:59]: yes, I will add this now

File: backend/ontario/ontario/__init__.py (hunk 2 above)
Comment (parkern342) [reply backend/ontario/ontario/__init__.py:125]: Composite does not currently exist as an operation, the operation gegl:layer does a similar job, but takes arguments described here: https://gegl.org/operations/gegl-layer.html. I will begin implementing this ASAP

File: gimp/pictonode-gimp-plugin/GIMP 2.99/pictonode/custom_nodes.py (hunk 3 above)
Comment (parkern342) [reply gimp/pictonode-gimp-plugin/GIMP 2.99/pictonode/custom_nodes.py:74]: oops didnt see this get included in changes

**Review by notgull — APPROVED** [review]
LGTM, merge when you're ready

**Pull request merged**